File: test-suite/src/test/java/io/micronaut/security/docs/authenticationparam/UserController.java
package io.micronaut.security.docs.authenticationparam;

//tag::imports[]
import io.micronaut.core.util.CollectionUtils;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.authentication.Authentication;
import org.jspecify.annotations.Nullable;

import java.util.Collections;
import java.util.Map;
//end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "AuthenticationParamTest")
//tag::clazz[]

@Controller("/user")
public class UserController {

    @Secured("isAnonymous()")
    @Get("/myinfo")
    public Map<String, Object> myinfo(@Nullable Authentication authentication) {
        if (authentication == null) {
            return Collections.singletonMap("isLoggedIn", false);
        }
        return CollectionUtils.mapOf("isLoggedIn", true,
                "username", authentication.getName(),
                "roles", authentication.getRoles()
        );
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/security/docs/authenticationparam/UserController.py
# tag::imports[]
from micronaut.http.annotation import Controller, Get
from micronaut.security.annotation import Secured
from micronaut.security.authentication import Authentication
# end::imports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="AuthenticationParamTest")
# tag::clazz[]

@Controller("/user")
class UserController:

    @Secured("isAnonymous()")
    @Get("/myinfo")
    def myinfo(self, authentication: Authentication | None) -> dict:
        if authentication is None:
            return {"isLoggedIn": False}
        return {"isLoggedIn": True,
                "username": authentication.getName(),
                "roles": list(authentication.getRoles())}
# end::clazz[]
